src/details/ArborX_DetailsConcepts.hpp

```diff
@@ -30,6 +30,33 @@ namespace ArborX
 namespace Details
 {
 
+struct not_a_type
+{
+  not_a_type() = delete;
+  ~not_a_type() = delete;
+  not_a_type(not_a_type const &) = delete;
+  void operator=(not_a_type const &) = delete;
+};
+
+// primary template handles all types not supporting the archetypal Op
+template <class, template <class...> class Op, class... Args>
+struct is_detected_impl : std::false_type
+{
+  using type = not_a_type;
+};
+
+// specialization recognizes and handles only types supporting Op
+template <template <class...> class Op, class... Args>
+struct is_detected_impl<std::void_t<Op<Args...>>, Op, Args...> : std::true_type
```

Comment:
How does this work? According to [this](https://en.cppreference.com/w/cpp/types/void_t) `void_t` is `c++17`

Reply:
https://github.com/arborx/ArborX/blob/19e2778e419cd7bedbb90d0ec9b9677f2ee51703/src/details/ArborX_DetailsConcepts.hpp#L19-L25